prusti-viper/src/encoder/type_encoder.rs

```diff
@@ -231,6 +231,59 @@ impl<'p, 'v, 'r: 'v, 'tcx: 'v> TypeEncoder<'p, 'v, 'tcx> {
         }
     }
 
+    fn is_ghost_adt(ghost_adt_def: &ty::AdtDef, item_name: String) -> Option<String> {
+        // check if Crate: prusti_contracts and Module: ghost
+        let item_name: Vec<&str> = item_name.split("::").collect();
+        let crate_name = item_name[0];
+        let mod_name = item_name[1];
+        let adt_identifier = item_name[2];
+        if crate_name.eq("prusti_contracts") && mod_name.contains("ghost"){
+            let ghost_name = ghost_adt_def.non_enum_variant().ident.as_str();
+            if ghost_name.contains("Ghost") {
+                return Some(ghost_name.to_string());
+            }
+        }
+        None
+    }
+
+    fn encode_ghost_predicate(ghost_type: &str, value_field: vir::Field) -> Vec<vir::Predicate> {
+        match ghost_type {
+            "GhostInt" => vec![vir::Predicate::new_primitive_value(
```

Comment:
I think it would be easiest to encode `GhostInt` in the same way as, for example, `u32` is encoded.

Reply:
> Do you suggest changing the name argument of the `vir::Predicate::new_primitive_value()` method to the type name instead of `vir::Type::GhostInt`?

Yes, see these lines of the `bool` encoding:

https://github.com/viperproject/prusti-dev/pull/181/files/33ac0c3e21d79636a7a595dc3eb9436aef2c7b14#diff-9d1fe0d43f4aa745d82726fcc766cfa7R291-R295